components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/processors/SPInitLogoutRequestProcessor.java

```diff
@@ -111,6 +111,10 @@ public SAMLSSOReqValidationResponseDTO process(LogoutRequest logoutRequest, Stri
                     .getPersistenceManager();
             String sessionIndex = logoutRequest.getSessionIndexes().size() > 0 ? logoutRequest
                     .getSessionIndexes().get(0).getSessionIndex() : null;
+            if (sessionIndex == null) {
```

Comment:
please add a code comment explaining why this is needed and what is being done here...

Reply:
fixed